addons/point_of_sale/models/pos_order.py

```diff
@@ -918,8 +919,19 @@ def _order_line_fields(self, line, session_id=None):
                 # fallback on any pos.order.line sequence
                 line[2]['name'] = self.env['ir.sequence'].next_by_code('pos.order.line')
 
+        if not Product.browse(line[2]['product_id']).exists():
+            _logger.warning("Order created with product %s but no longer exists")
+            product = self.env.ref('point_of_sale.product_unknown', raise_if_not_found=False)
+            if not product:
+                line[2]['product_id'] = self.env['ir.model.data'].sudo()._update('product.product', 'point_of_sale', {
+                    'name': _("Missing Product"),
+                    'active': False,
+                }, xml_id='product_unknown')
+            else:
+                line[2]['product_id'] = product.id
+
         if line and 'tax_ids' not in line[2]:
-            product = self.env['product.product'].browse(line[2]['product_id'])
+            product = Product.browse(line[2]['product_id'])
             line[2]['tax_ids'] = [(6, 0, [x.id for x in product.taxes_id])]
```

Comment:
Taxes should be preserved from the POS data, isn't it?

Reply:
I am not sure to understand your comment. My code does not change this behaviour.
This `if` is the scenario where tax_ids is missing, we try to find it from the product. If there is no tax_ids *and* a missing product, then we will get the taxes of the fallback product (probably none)

Reply:
Actually I am not sure why this `if ` is present as we should always have a `tax_ids` key given in the `create_from_ui` call. In the scenario you are giving, I have deleted `product_id` but a `tax_ids` referring to the 10% so will match the scenario. I tested, got the right tax even if my rescue product has no tax.

Reply:
OK, I see that the `if` condition only activates when no tax is found. This should come from a very corner case when taxes were not passed, and maybe it's even dead code. Thanks for checking.